monai/networks/layers/factories.py

```diff
@@ -68,34 +68,36 @@ def use_factory(fact_args):
 import torch.nn as nn
 
 from monai.networks.utils import has_nvfuser_instance_norm
-from monai.utils import look_up_option, optional_import
+from monai.utils import ComponentStore, look_up_option, optional_import
 
 __all__ = ["LayerFactory", "Dropout", "Norm", "Act", "Conv", "Pool", "Pad", "split_args"]
 
 
-class LayerFactory:
+class LayerFactory(ComponentStore):
     """
     Factory object for creating layers, this uses given factory functions to actually produce the types or constructing
     callables. These functions are referred to by name and can be added at any time.
     """
 
-    def __init__(self) -> None:
-        self.factories: dict[str, Callable] = {}
-
-    @property
-    def names(self) -> tuple[str, ...]:
+    def add_factory_callable(self, name: str, func: Callable, desc: str | None = None) -> None:
         """
-        Produces all factory names.
+        Add the factory function to this object under the given name, with optional description.
         """
+        description: str = desc or func.__doc__ or ""
+        self.add(name.upper(), description, func)
+        self.__doc__ = (
```

Comment:
This is going to overwrite the docstring you've already written, maybe the last line of the docstring should be "Please see :py:class:`monai.networks.layers.split_args` for additional args parsing.\n\nThe supported member(s) are:" then append new  lines when something is added with this method.

Reply:
adressed so it now appends. unfortunately needs an assert before the append to stop mypy throwing errors!